Refit/FormValueDictionary.cs

```diff
@@ -45,15 +51,74 @@ public FormValueDictionary(object source, RefitSettings settings)
                     if (value != null)
                     {
+                        var fieldName = GetFieldNameForProperty(property);
+
                         // see if there's a query attribute
                         var attrib = property.GetCustomAttribute<QueryAttribute>(true);
 
-                        Add(GetFieldNameForProperty(property), settings.FormUrlEncodedParameterFormatter.Format(value, attrib?.Format));
+                        if (value is IEnumerable enumerable) {
+                            switch (attrib?.CollectionFormat) {
+                                case CollectionFormat.Multi:
+                                    foreach (var item in enumerable) {
+                                        Add(fieldName, settings.FormUrlEncodedParameterFormatter.Format(item, attrib.Format));
+                                    }
+                                    break;
+                                case CollectionFormat.Csv:
+                                case CollectionFormat.Ssv:
+                                case CollectionFormat.Tsv:
+                                case CollectionFormat.Pipes:
+                                    var delimiter = attrib.CollectionFormat == CollectionFormat.Csv ?  ","
+                                        : attrib.CollectionFormat == CollectionFormat.Ssv ? " "
+                                        : attrib.CollectionFormat == CollectionFormat.Tsv ? "\t" : "|";
+
+                                    var formattedValues = enumerable
+                                        .Cast<object>()
+                                        .Select(v => settings.FormUrlEncodedParameterFormatter.Format(v, attrib.Format));
+                                    Add(fieldName, string.Join(delimiter, formattedValues));
+                                    break;
+                                default:
+                                    Add(fieldName, settings.FormUrlEncodedParameterFormatter.Format(value, attrib?.Format));
+                                    break;
+                            }
+                        }
+                        else
+                        {
+                            Add(fieldName, settings.FormUrlEncodedParameterFormatter.Format(value, attrib?.Format));
+                        }
+
                     }
                 }
             }
         }
 
+        /// <summary>
+        /// Returns a key for each entry. If multiple entries share the same key, the key is returned multiple times.
+        /// </summary>
+        public IEnumerable<string> Keys => this.Select(it => it.Key);
+
+        /// <summary>
+        /// Returns the value of the first entry found with the matching key. Multiple additional entries may use the
+        /// same key, but will not be considered. Returns <c>null</c> if no matching entry is found.
+        /// </summary>
+        public string this[string key]
+        {
```

Comment:
Where is this currently used? Is there a danger in losing in this path that needs to be addressed? In other words, should this indexer be removed and coded around a different way?

Reply:
Existing tests use it. I think there is a larger than necessary risk of future misuse, and the tests can be satisfied comfortably with a LINQ expression. Fixed in c63b62c7164a70a32fa74c7dcfbd50fd4b23b534